### PR TITLE
Add a simple bash test

### PR DESCRIPTION
**Pull request by sreya92**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -64,6 +64,8 @@ func main() {
 		sshFlags     = flag.String("ssh-flags", "", "custom SSH flags")
 		syncBack     = flag.Bool("b", false, "sync extensions back on termination")
 		printVersion = flag.Bool("version", false, "print version information and exit")
+		port         = flag.String("port", "", "Start VS Code on the provided port. If one is not provided a random one is selected")
+		noOpen       = flag.Bool("no-open", false, "Start VS Code but don't open a Chrome app window")
```
Comment (nhooyr) [inline main.go]: lets use $DISPLAY = "" for this

Comment (ammario) [reply main.go]: Is `DISPLAY` set on all macs?

File: test.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+done
+
+if [ "$ATTEMPTS" -eq "$MAX_ATTEMPTS" ];
+then
+  echo "Timed out waiting for code server to start"
+  exit 1
+fi
+
+# Curl VS Code remotely.
+ssh "$REMOTE_IP" "curl -o /dev/null --connect-timeout 5 -s localhost:$VS_CODE_PORT/"
```
Comment (ammario) [inline test.sh]: After reading this I feel like we're going to have to do this in Go as soon as we add any more complexity.

File: main.go (hunk 1 above)
Comment (nhooyr) [reply main.go]: Yes, similar to how there is an X11 socket whose path is given by $DISPLAY, $DISPLAY on macOS contains the quartz socket.

Comment (ammario) [reply main.go]: What about WSL?

Comment (nhooyr) [reply main.go]: WSL is almost certainly using some sort of X11 forwarding so it would have $DISPLAY.

File: sshcode_test.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+			return
+		}
+		err = backoff.Wait(ctx)
+		require.NoError(t, err)
+	}
+}
+
+// fakeRSAKey isn't used for anything other than the trashh ssh
+// server.
+const fakeRSAKey = `-----BEGIN RSA PRIVATE KEY-----
```
Comment (scsmithr) [inline sshcode_test.go]: bruh, why you exposing my private key?

Comment (nhooyr) [reply sshcode_test.go]: unforgivable.

File: main.go [hunk 4]
```diff
@@ -64,6 +65,8 @@ func main() {
 		sshFlags     = flag.String("ssh-flags", "", "custom SSH flags")
 		syncBack     = flag.Bool("b", false, "sync extensions back on termination")
 		printVersion = flag.Bool("version", false, "print version information and exit")
+		remotePort   = flag.String("remote-port", "", "Start VS Code on the provided port. If one is not provided a random one is selected")
```
Comment (nhooyr) [inline main.go]: lets not expose these to users.

File: main.go [hunk 5]
```diff
@@ -103,66 +106,96 @@ Arguments:
...
-	localPort, err := randomPort()
+
+	if o.localPort == "" {
+		o.localPort, err = randomPort()
+	}
+	if err != nil {
+		return xerrors.Errorf("failed to find available local port: %w", err)
+	}
+
+	// TODO pick a random remote port
```
Comment (nhooyr) [inline main.go]: Confusing comment

File: sshcode_test.go [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+			localPort:  localPort,
+			remotePort: remotePort,
+		})
+		require.NoError(t, err)
+	}()
+
+	waitForSSHCode(t, localPort, time.Second*30)
+	waitForSSHCode(t, remotePort, time.Second*30)
+
+	out, err := exec.Command("pkill", "codessh").CombinedOutput()
```
Comment (nhooyr) [inline sshcode_test.go]: is this necessary.

Comment (sreya92) [reply sshcode_test.go]: sshcode won't exit otherwise as it waits for a sigterm or the long-running ssh process to exit

Comment (nhooyr) [reply sshcode_test.go]: Ah. Wouldn't you want `os.Pid` here with `kill`? the go test binary can have a weird name.

Comment (sreya92) [reply sshcode_test.go]: Nah this doesn't kill the `go test` process this kills the `code-server` process. Tests that sshcode exits gracefully when the `coder-server` process is terminated

Comment (nhooyr) [reply sshcode_test.go]: Why not `pkill code-server`? Why `codessh`?

Comment (sreya92) [reply sshcode_test.go]: The process is called `codessh`, when we start it

Comment (sreya92) [reply sshcode_test.go]: Honestly having it called `codessh` is pretty confusing, there's nothing different about it to warrant a different name

Comment (nhooyr) [reply sshcode_test.go]: I'm confused, I don't see a `codessh` anywhere.

Comment (sreya92) [reply sshcode_test.go]: https://github.com/cdr/sshcode/blob/test/main.go#L133

Comment (nhooyr) [reply sshcode_test.go]: mm got it. def a good idea to rename, very confusing.

**Pull request merged**